### ISSUE TITLE
Invalid warning for `if block ends with a return statement, so drop this else...`

### ISSUE THREAD

**Issue by pengux** [golang/lint#373]
An example code:

```
        if v, ok := foobarMap["foo"]; !ok {
		return errors.New("foo not found")
	} else {
		hello = v
	}
```

This gives a warning but the `else` block is necessary here as the `v` variable is scoped

**Comment by syoder** [comment]
I keep running into this too.

**Comment by devdinu** [comment]
You can reformat this as
```
var v interface{}
v, ok := foobarMap["foo"]
if !ok {
     return errors.New("foo not found")
}
hello = v
``` 
and would be more idiomatic.

**Comment by svenkanna** [comment]
I too keep running into this issue and it is very annoying every time to modify the code as mentioned in the suggestion. The purpose of having variables local to the if-else block is defeated here. The warning is valid only if none of the variables declared in an if statement are not used in the else block.

**Comment by t2wu** [comment]
@svenkanna I agree with you about the variable local to the if else. This drives me nuts. Golang specific idiomatic issue aside, many times I find the relationship between the if-else more specific when the else block is present and I consider it a better style. 

Is there an option to stop bugging me about this? I find the linter too opinionated.

**Comment by andrewwatkin** [comment]
I agree with almost everything that the linter reports, except this warning. There are many situations where code is clearer if the symmetry between the 'if' and 'else' blocks is preserved. Eliminating the 'else' does not always make for clearer code.

**Comment by mvdan** [comment]
Thank you for submitting this issue! As per https://github.com/golang/go/issues/38968, we are freezing and deprecating golint. There's no drop-in replacement to golint per se, but you should find that [Staticcheck](https://staticcheck.io/) works well in encouraging good Go code, much like golint did in the past, since it also includes style checks. There's always `gofmt` and `go vet` too, of course.

If you would like to contribute further, I'd encourage you to engage [Staticcheck's issue tracker](https://github.com/dominikh/go-tools/issues) or look at [vet's open issues](https://github.com/golang/go/issues?q=is%3Aissue+is%3Aopen+cmd%2Fvet+in%3Atitle), as they are both actively maintained. If you have an idea that doesn't fit into either of those tools, you could look at [other Go linters](https://github.com/golangci/awesome-go-linters), or write your own - these days it's fairly straightforward with [go/analysis](https://pkg.go.dev/golang.org/x/tools/go/analysis).

To help avoid confusion, I'm closing all issues before we freeze the repository. If you have any feedback, you can leave a comment on the proposal thread where it was decided to deprecate golint - though note that the proposal has been accepted for nearly a year. Thanks!
